internal/engine/netx/quicdialer/connectionstate.go

```diff
@@ -10,5 +8,5 @@ import (
 
 // ConnectionState returns the ConnectionState of a QUIC Session.
 func ConnectionState(sess quic.EarlySession) tls.ConnectionState {
-	return tls.ConnectionState{}
+	return sess.ConnectionState().TLS.ConnectionState
```

Comment:
v0.20.x of lucas-clemente/quic-go moved `.ConnectionState` inside a `.TLS` subobject.

The function definition is now:

```Go
type Session interface {
	// ...
	ConnectionState() ConnectionState
}
```

In turn, `ConnectionState` is defined as

```Go
type ConnectionState struct {
	TLS               handshake.ConnectionState
	SupportsDatagrams bool
}
```

and `handshake.ConnectionState` is defined as

```Go
type ConnectionState = qtls.ConnectionState
```

and `qtls.ConnectionState` is

```Go
type ConnectionState = qtls.ConnectionStateWith0RTT
```

and `qtls.ConnectionStateWith0RTT` is

```Go
type ConnectionStateWith0RTT struct {
	ConnectionState
	Used0RTT bool // true if 0-RTT was both offered and accepted
}
```

and `ConnectionState` is

```Go
type ConnectionState = tls.ConnectionState
```

and `tls.ConnectionState` is the real `tls.ConnectionState`, which is not an interface.

Because none of these is an interface, the above code is fine: `TLS` cannot be `nil` and `ConnectionState` cannot be `nil`. (The issue we have here is that `sess` could be `nil` but this means the caller has called `ConnectionState()` on the wrong object &mdash; so I think all of this is fine.)